### DISCUSSION TITLE
Error loading System.Data.SqlClient when using .net core runner

### DISCUSSION THREAD

**Discussion by achikhv** [cake-build/cake#3097] [Q&A] [answered]
### What You Are Seeing?
I am using cake runner for .net framework and considering migrating to .net core runner. My build scripts uses SqlConnection from System.Data.SqlClient to execute some SQL.
It works when .NET Framework runner is used. But switching to .NET Core causes error:
```
        Unable to load DLL 'sni.dll' or one of its dependencies: The specified module could not be found. (0x8007007E)
```

### What is Expected?
SqlClient and all of its dependencies are loaded without errors.

### What version of Cake are you using?
0.38.5

### Are you running on a 32 or 64 bit system?
64-bit

### What environment are you running on?  Windows? Linux? Mac?
Windows

### Are you running on a CI Server?  If so, which one?
No

### How Did You Get This To Happen? (Steps to Reproduce)
1. `dotnet new tool-manifest`
2. `dotnet tool install Cake.Tool --version 0.38.5`
3. 
```
#addin System.Data.SqlClient&version=4.5.0

Task("Default")
.Does(() =>
{
	using var connection = new System.Data.SqlClient.SqlConnection("Data Source=(localdb)\\MsSqlLocalDb; Integrated Security=true");

	connection.Open();

	Information("got here");
});

RunTarget(Argument("Target", "Default"));
```
4. `dotnet cake`

### Output Log
https://gist.github.com/achikhv/95539747148cd3e6e35b7a4e8a6da4de

**Comment by augustoproiete** [comment, marked as answer]
@achikhv The issue is that `System.Data.SqlClient` for .NET Core depends on other NuGet packages that need to be downloaded as well by Cake. If you expand the "[Dependencies](https://www.nuget.org/packages/System.Data.SqlClient#show-dependency-groups)" accordion on nuget.org you see something like this: 

[![image](https://user-images.githubusercontent.com/177608/107180512-3a1b6800-69af-11eb-9445-e83b536c5689.png)](https://www.nuget.org/packages/System.Data.SqlClient#show-dependency-groups)

In the .NET Framework world these dependencies already exist on your Global Assembly Cache (GAC), which is why it works today.

Cake currently does not download dependencies by default, so you have to be explicit about it by adding a [`&loaddependencies=true`](https://cakebuild.net/docs/writing-builds/preprocessor-directives#dependencies) to the addin preprocessor directive:

```csharp
#addin System.Data.SqlClient&version=4.5.0&loaddependencies=true

// ...
```

The above should resolve the error you're seeing.

---

ps:

- `System.Data.SqlClient` has been superseded by [`Microsoft.Data.SqlClient`](https://www.nuget.org/packages/Microsoft.Data.SqlClient/). You might want to consider using that instead. For more info: "[Introducing the new Microsoft.Data.SqlClient](https://devblogs.microsoft.com/dotnet/introducing-the-new-microsoftdatasqlclient/)".

- In case you haven't seen it, [we released Cake v1.0.0 earlier today](https://cakebuild.net/blog/2021/02/cake-v1.0.0-released) so you might want to take this opportunity to also upgrade to the latest version of Cake.

**Reply by achikhv** [reply]
@augustoproiete , thanks for the answer! Adding `loaddependencies=true` did the job. Somehow I missed this piece of docs and could not figure out how to force loading dependencies. 

Thanks for the note about Cake v1.0. I have seen the release, and decided to migrate to dotnet runner first. Next step will definitely be Cake v1.0.
